swagger.yaml

```diff
@@ -0,0 +1,333 @@
+    delete:
+      tags:
+      - "Filter"
+      summary: "Remove a dimension and any values set within the dimension"
+      description: "Remove a dimension and any values set within the dimension"
+      responses:
+        200:
+          description: "The dimension was removed"
+        400:
+          description: "The filter job has been locked as it has been `submitted` to be processed"
+        404:
+          $ref: '#/responses/FilterOrdimensionNotFound'
+        500:
+          $ref: '#/responses/InternalError'
+    get:
+      tags:
+      - "Filter"
+      summary: "Get all values from a dimension which have been set"
+      description: "Get a list of all values which will be used to filter the dimension"
+      responses:
+        200:
+          description: "A list of all values for a dimension was returned"
+          schema:
+            $ref: '#/definitions/Dimension'
+        400:
+          description: "The filter job has been locked as it has been `submitted` to be processed"
+        404:
+          $ref: '#/responses/FilterOrdimensionNotFound'
+        500:
+          $ref: '#/responses/InternalError'
+  /filters/{filter_id}/dimensions/{dimension_name}/options/{value}:
+    parameters:
+      - $ref: '#/parameters/filter_id'
+      - $ref: '#/parameters/dimension_name'
+      - $ref: '#/parameters/value'
+    get:
+      tags:
+      - "Filter"
+      summary: "Check if a value exists within a dimension"
+      description: "Check if a value exists within a dimension"
+      responses:
+        200:
+          description: "Value exists within the dimension"
+        400:
+          $ref: '#/responses/FilterOrdimensionNotFound'
+        404:
+          description: "No Value found in dimension"
+        500:
+          $ref: '#/responses/InternalError'
+    post:
+      tags:
+      - "Filter"
+      summary: "Add a value to a dimension to filter on"
+      description: "Add a value to a dimension to filter on"
+      responses:
+        200:
+          description: "Value was added"
+        400:
+          description: "The filter job has been locked as it has been `submitted` to be processed"
+        404:
+          $ref: '#/responses/FilterOrdimensionNotFound'
+        500:
+          $ref: '#/responses/InternalError'
+    delete:
+      tags:
+      - "Filter"
+      summary: "Remove a value from a dimension"
+      description: "Remove a single value from a dimension"
+      responses:
+        200:
+          description: "Value was removed"
+        400:
+          description: "The filter job has been locked as it has been `submitted` to be processed"
+        404:
+          $ref: '#/responses/FilterOrdimensionNotFound'
+        500:
+          $ref: '#/responses/InternalError'
+responses:
+  FilterJobNotFound:
+    description: "Filter Job not found"
+  FilterOrdimensionNotFound:
+    description: "The filter job has been locked as it has been `submitted` to be processed"
```

Comment:
Is this the right message? I was expecting it to just be something like 'Dimension not found.' A number of the endpoints are returning the same message for 400 and 404 error?